app/models/post.rb

```diff
@@ -0,0 +1,12 @@
+class Post < ApplicationRecordhas_many :likes, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+
+  def update_counter(count, user_id)
+    User.find(user_id).update(posts_counter: count)
+  end
```

Comment:
- [ ] You do not need to pass count and user_id on this method. Since post belongs to a author(user) you can get the post author without finding it on the database. Please fix this issue. You can consider using this code `author.increment!(:posts_counter)`.

Reply:
Hi @thecodechaser you right, I have remove it